### PR TITLE
Converts the remaining templates to Jinja2.

### PR DESCRIPTION
**Pull request by rossjones**
Unfortunately needs helpers to work around shortcomings of list-comps in jinja templates.

### CODE REVIEW

**Comment by davidread** [comment]
At /harvest the inactive sources are showing when you display the page, and they shouldn't. I looked briefly look, but I couldn't see why.

File: ckanext/harvest/templates/index.html [hunk 1]
```diff
@@ -27,78 +21,95 @@
...
-              <h3>All organizations</h3>
-              ${harvest_source_table(c.sources, can_edit)}
+{% endblock%}
 
-            </py:when>
-            <py:otherwise>
-                <div id="no-harvest-sources">No harvest sources defined yet.</div>
-            </py:otherwise>
-          </py:choose>
+{% block optional_footer %}
```
Comment (davidread) [inline ckanext/harvest/templates/index.html]: BTW I can't find optional_footer in the jinja templates - I think this should just be moved to the bottom of the block above it.

**Comment by davidread** [comment]
The rest looks fine - just the inactive ones for you to fix, if that's ok.

**Comment by davidread** [comment]
Deployed to prod2 & 3

**Comment by davidread** [comment]
All merged and deployed.

However there are some remaining ones in ckanext-dgu:

```
dread@dmbp:~/v-3/src$ find . -name "*.html" | grep -v failover |grep -v ckan/templates_legacy | grep -v ckanext/stats/templates_legacy | xargs grep genshi
./ckanext-dgu/ckanext/dgu/theme/templates/_dgu_util.html:  xmlns:i18n="http://genshi.edgewall.org/i18n"
./ckanext-dgu/ckanext/dgu/theme/templates/_dgu_util.html:  xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/group/edit.html:<html xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/group/new.html:<html xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/layout_base.html:      xmlns:i18n="http://genshi.edgewall.org/i18n"
./ckanext-dgu/ckanext/dgu/theme/templates/layout_base.html:      xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/layout_base.html:      xmlns:i18n="http://genshi.edgewall.org/i18n"
./ckanext-dgu/ckanext/dgu/theme/templates/layout_base.html:      xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/layout_base.html:  </div> <!-- /footer genshi -->
./ckanext-dgu/ckanext/dgu/theme/templates/publisher/apply.html:<html xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/publisher/users.html:<html xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/revision/diff.html:<html xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/user/login.html:<html xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/user/logout.html:<html xmlns:py="http://genshi.edgewall.org/"
./ckanext-dgu/ckanext/dgu/theme/templates/user/logout_first.html:<html xmlns:py="http://genshi.edgewall.org/"
'''
```

**Comment by rossjones** [comment]
Could have sworn we'd done those already :( 

layout_base and _dgu_util are to be deleted once everything is done.

**Comment by davidread** [comment]
Remaining work to be covered in https://github.com/datagovuk/ckanext-dgu/issues/313

**Pull request merged**